nifi-extension-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/main/java/org/apache/nifi/oauth2/JWTBearerOAuth2AccessTokenProvider.java

```diff
@@ -0,0 +1,677 @@
+            .expressionLanguageSupported(ExpressionLanguageScope.ENVIRONMENT)
+            .build();
+
+    public static final PropertyDescriptor SUBJECT = new PropertyDescriptor.Builder()
+            .name("Subject")
+            .description("The subject claim (sub) for the JWT.")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.ENVIRONMENT)
+            .build();
+
+    public static final PropertyDescriptor AUDIENCE = new PropertyDescriptor.Builder()
+            .name("Audience")
+            .description("The audience claim (aud) for the JWT. Space-separated list of audiences if multiple are expected.")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.ENVIRONMENT)
+            .build();
+
+    public static final PropertyDescriptor SCOPE = new PropertyDescriptor.Builder()
+            .name("Scope")
+            .description("The scope claim (scope) for the JWT.")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.ENVIRONMENT)
+            .build();
+
+    public static final PropertyDescriptor GRANT_TYPE = new PropertyDescriptor.Builder()
+            .name("Grant Type")
+            .description("Value to set for the grant_type parameter in the request to the token endpoint.")
+            .required(true)
+            .defaultValue("urn:ietf:params:oauth:grant-type:jwt-bearer")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.ENVIRONMENT)
+            .build();
+
+    public static final PropertyDescriptor ASSERTION = new PropertyDescriptor.Builder()
+            .name("Assertion Parameter Name")
+            .description("Name of the parameter to use for the JWT assertion in the request to the token endpoint.")
+            .required(true)
+            .defaultValue("assertion")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.ENVIRONMENT)
+            .build();
+
+    public static final PropertyDescriptor NBF = new PropertyDescriptor.Builder()
+            .name("Set Not Before Time")
+            .description("The nbf (not before) claim identifies the time before which the JWT MUST NOT be accepted for processing."
+                    + " If set to true, the current time will be used.")
+            .required(true)
+            .allowableValues("true", "false")
+            .defaultValue("false")
+            .addValidator(Validator.VALID)
+            .build();
+
+    public static final PropertyDescriptor HEADER_TYPE = new PropertyDescriptor.Builder()
+            .name("Set JWT Header Type")
```

Comment:
Is there any reason not to set this to `JWT` for all tokens as described in [RFC 7519](https://www.rfc-editor.org/rfc/rfc7519#section-5.1)?

Reply:
Same answer: some implementations are very strict in terms of expected fields in the header. I will test again to double check to be sure.

Reply:
My bad, Salesforce seems to be OK when the header type is set, so I'll remove this property